File: app/src/main/java/com/kanj/apps/topbanner/core/InvalidBannerOverlayException.java
package com.kanj.apps.topbanner.core;

/**
 * Created by Kanj Narayan on 22/08/17.
 */

/**
 * Exception thrown when trying to add banner to a BannerProgressAbstractActivity subclass which
 * does not provide a FrameLayout container for banner overlay.
 */
public class InvalidBannerOverlayException extends Exception {
    private static final String MSG = "Banner overlay container is null or not a FrameContainer";
    public InvalidBannerOverlayException() {
        super(MSG);
    }
}

File: app/src/main/java/com/kanj/apps/topbanner/core/InvalidBannerOverlayException.kt
package com.kanj.apps.topbanner.core

/**
 * Exception thrown when trying to add banner to a BannerProgressAbstractActivity subclass which
 * does not provide a FrameLayout container for banner overlay.
 */
class InvalidBannerOverlayException : Exception("Banner overlay container is null or not a FrameContainer") {
}